src/Core/Scheduler/Providers/GpxUtils.js

```diff
@@ -16,78 +19,172 @@ function _gGpxToWTrackPointsArray(gpxXML) {
+            mesh.onBeforeRender = updatePointScale;
+
+            points.add(mesh);
         }
-        const points = new THREE.Points();
-        points.geometry.addAttribute('position', new THREE.BufferAttribute(positions, 3));
-        points.material.color.setRGB(0, 1, 0);
         return points;
     } else {
         return null;
     }
 }
 
-function _gpxToWTrackPointsMesh(gpxXML) {
+function updatePath(renderer, scene, camera) {
+    const distance = getDistance(this, camera);
+    this.material.depthTest = distance < this.geometry.boundingSphere.radius * 2;
+    const size = renderer.getSize();
+    this.material.uniforms.resolution.value.set(size.width, size.height);
+}
+
+function _gpxToWTrackPointsMesh(gpxXML, options) {
     var trackPts = _gGpxToWTrackPointsArray(gpxXML);
 
     if (trackPts.length) {
-        const positions = new Float32Array(trackPts.length * 3);
-        for (var i = 0; i < trackPts.length; i++) {
-            const pos = _gpxPtToCartesian(trackPts[i]);
-            pos.toArray(positions, 3 * i);
+        gpxXML.center = gpxXML.center || _gpxPtToCartesian(trackPts[0], options.crs);
+
+        var geometry = new THREE.Geometry();
+
+        for (const trackPt of trackPts) {
+            const point = _gpxPtToCartesian(trackPt, options.crs).sub(gpxXML.center);
+            geometry.vertices.push(point);
+        }
+
+        var line = new Line.MeshLine();
+        line.setGeometry(geometry);
+        // Due to limitations in the ANGLE layer,
+        // with the WebGL renderer on Windows platforms
+        // lineWidth will always be 1 regardless of the set value
+        // Use MeshLine to fix it
+        var material = new Line.MeshLineMaterial({
+            lineWidth: options.lineWidth || 12,
+            sizeAttenuation: 0,
+            color: new THREE.Color(0xFF0000),
+        });
+
+        if (Capabilities.isLogDepthBufferSupported()) {
+            material.fragmentShader = material.fragmentShader.replace(/.*/, '').substr(1);
+            patchMaterialForLogDepthSupport(material);
+            // eslint-disable-next-line no-console
+            console.warn('MeshLineMaterial shader has been patched to add log depth buffer support');
         }
 
-        const line = new THREE.Line();
-        line.geometry.addAttribute('position', new THREE.BufferAttribute(positions, 3));
-        line.material.color.setRGB(1, 0, 0);
-        line.material.linewidth = 100;
+        const pathMesh = new THREE.Mesh(line.geometry, material);
+        // update size screen uniform
+        // update depth test for visibilty path, because of the proximity of the terrain and gpx mesh
+        pathMesh.onBeforeRender = updatePath;
 
-        return line;
+        return pathMesh;
     } else {
         return null;
     }
 }
 
-function _gpxToMesh(gpxXML) {
+function _gpxToMesh(gpxXML, options) {
     if (!gpxXML) {
         return undefined;
     }
 
+    options = options || {};
```

Comment:
You can remove this, if you make sure that `options` is always defined.
And the simplest way to do this is to fix the only entry point for this module:
```js
load(urlFile, crs, options = {}) {
...
```